### ISSUE TITLE
Invalid Project: Missing _springroll_usertasks Error

### ISSUE THREAD

**Issue by probityrules** [SpringRoll/SpringRollStudio#10]
We just installed SpringRollStudio on a Mac and are running into the same issue mentioned in https://github.com/SpringRoll/SpringRollStudio/issues/4. Here are the steps to reproduce:
1. Open SpringRollStudio
2. Click "New"
3. Set up a new project using the default template (v1.5.0)
4. Once it's created successfully, click "Tasks" in SpringRollStudio
5. The Tasks window reports "Installing plugins. This may take a few minutes." It takes a few minutes.
6. Changes to "Waiting for tasks. Hold on." for about 5 seconds.
7. Finally reports "Invalid project, must contain Grunt task '_springroll_usertasks'." in red.

I found that '_springroll_usertasks' is provided by https://github.com/SpringRoll/grunt-springroll-helper so I ran `npm install grunt-springroll-helper` and received the following logs:

```
@0.0.1 /Users/gopherwoodstudios/Desktop/Repositories/test-project
└─┬ grunt-springroll@0.7.10
  ├── UNMET PEER DEPENDENCY grunt@0.4.5
  └── grunt-springroll-helper@0.1.2 

npm WARN grunt-concat-json@0.0.10 requires a peer of grunt@~0.4.5 but none was installed.
npm WARN grunt-contrib-watch@0.6.1 requires a peer of grunt@~0.4.0 but none was installed.
npm WARN grunt-simple-version@0.2.4 requires a peer of grunt@^0.4.5 but none was installed.
npm WARN grunt-springroll-helper@0.1.2 requires a peer of grunt@~0.4.4 but none was installed.
```

So then I thought maybe `npm install grunt@0.4.5` might work and then ran `npm install grunt-springroll-helper` again. It cleared up the unmet peer dependency error, but that doesn't seem to make a difference on the SpringRollStudio Tasks error, as if "grunt-springroll-helper" is still not installed.

Any help would be appreciated!

**Comment by prlandwehr** [comment]
For the record I'm running into the same issue on my Windows 10 machine, However, my mac on osx 10.12 works fine.

No idea why.

**Comment by yanshcherbakovatredspace** [comment]
@probityrules and @prlandwehr You need to run `grunt _springroll_usertasks` manually when this happens. This is a known bug for SpringrollStudio. It should be running this automatically but doesn't seem to be doing so.

**Comment by prlandwehr** [comment]
Here is my console output from when I tried to run "grunt _springroll_usertasks" in the directory of the active project. So still no luck getting this working on windows.

>> Local Npm module "grunt-bower-task" not found. Is it installed?
>> Local Npm module "grunt-concat-json" not found. Is it installed?
>> Local Npm module "grunt-contrib-clean" not found. Is it installed?
>> Local Npm module "grunt-contrib-concat" not found. Is it installed?
>> Local Npm module "grunt-contrib-connect" not found. Is it installed?
>> Local Npm module "grunt-contrib-copy" not found. Is it installed?
>> Local Npm module "grunt-contrib-jshint" not found. Is it installed?
>> Local Npm module "grunt-contrib-less" not found. Is it installed?
>> Local Npm module "grunt-contrib-uglify" not found. Is it installed?
>> Local Npm module "grunt-contrib-watch" not found. Is it installed?
>> Local Npm module "grunt-newer" not found. Is it installed?
>> Local Npm module "grunt-simple-version" not found. Is it installed?
>> Local Npm module "grunt-springroll-helper" not found. Is it installed?
>> Local Npm module "grunt-text-replace" not found. Is it installed?
Warning: Task "_springroll_usertasks" not found. Use --force to continue.

Aborted due to warnings.

**Comment by mkucic** [comment]
@prlandwehr Did you ever find a solution to this? I'm running into the same issue.

**Comment by andrewstart** [comment]
I think that I figured this out on another project elsewhere - a (largely inactive) dependency needs to update one of its dependencies to support npm 3+. 
A potential workaround is adding the 'missing' projects to your package.json with a different version from what grunt-springroll looks for. That should force npm to install another copy of those dependencies within grunt-springroll instead of only at the top level.

**Comment by mkucic** [comment]
An addendum to the workaround posted by @andrewstart 

I've found that directly copying the mentioned dependencies from grunt-springroll to the project's dependencies also works. Keeping the same versions seemed to work just as well.

**Comment by vikramarka** [comment]
Best way to fix this is to run npm install inside \node_modules\grunt-springroll.

**Comment by chipbell4** [comment]
Hi folks, this is actually a bug in `grunt-springroll`, that has just been resolved with a new patch update from this PR https://github.com/SpringRoll/grunt-springroll/pull/6 and is available in [grunt-springroll 0.7.11](https://www.npmjs.com/package/grunt-springroll). Let me know if you're still running into this issue
